File: Data Structures & Algorithms/partition-equal-subset-sum/submission-2.cpp
class Solution {
public:
    bool canPartition(vector<int>& nums) {
        int s = accumulate(nums.begin(), nums.end(), 0);
        if(s%2 == 1)
            return false;
        
        int n = s / 2;

        vector<bool> dp(n + 1, false);
        dp[0] = true;

        for(int i : nums) {
            for(int j = n; j >= i; j--) {
                if(dp[j - i])
                    dp[j] = true;
            }
            if(dp[n])
                return true;
        }
        return false;
    }
};

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for idx, n in enumerate(nums):
            if target - n in d:
                return [d[target-n], idx]
            d[n] = idx
        return [-1, -1]